learningequality/ka-lite
# Hot fix for failing video downloads on OSX.

Hi @aronasorman, this PR does the following:
* fixes #3704 
* fixes #4114 
* fixes #3990 

As per learningequality/installers#100 - PyRun doesn't include the `multiprocessing` module by default.  So we check if `call_command()` is being run inside PyRun and call another method that uses the `subprocess` module.  

If not running on Pyrun, we resume with current functionality that calls the `CommandProcess` class which uses the `multiprocessing` module.

**To test this with Pyrun**

1. Checkout the [installers repo](https://github.com/learningequality/installers) on another directory, then run `setup.sh` of the OSX installers repo.
2. Press Ctrl+C or cancel when it's already downloading the "ka-lite" repo, we only need the downloaded PyRun package.
3. Checkout this PR's branch.
4. Run `"<installers-working-dir>/installers/osx/temp/pyrun-2.7/bin/pip" install .` to install the `ka-lite-static` module.
5. Run `"<working-dir>/installers/osx/temp/pyrun-2.7/bin/pip" freeze` - to see double-check that `ka-lite-static` is installed.
6. Run `launchctl setenv KALITE_PYTHON "<bin/pyrun full path>"`.  Example: `launchctl setenv KALITE_PYTHON "/Users/cyril/w/fle/ka-lite/installers/osx/temp/pyrun-2.7/bin/pyrun"`
7. Restart the terminal.
8. Check that we have a `KALITE_PYTHON` env var with `env | grep KA`.
9. You can now run `bin/kalite manage setup` command and test the device registration, video downloads, language downloads, and etc to see if they work.

To test with the OSX default Python, just unset the `KALITE_PYTHON` with `launchctl unsetenv KALITE_PYTHON` and restart the server.

**Tips**
1. Make an alias of the 'pyrun' and 'pip' to ease typing.
2. Set DEBUG=True in your `local_settings.py` for easier tracing.

**Other fix**
This also sets the default value for the `DO_NOT_RELOAD_CONTENT_CACHE_AT_STARTUP = False` at `kalite/settings/__init__.py` to fix crashes when `settings.DO_NOT_RELOAD_CONTENT_CACHE_AT_STARTUP` is used in other Python codes as per #4114.  I had to do this to continue testing the fixes above.
